Establish structure and process for in-repo release notes (#18427)
bevyengine/bevy: working-release-notes/README.md

# Bevy's Release Notes Process

Hi! Did someone add `M-Needs-Release-Note` to your PR? If so, you're in the right place.
Let's talk about how this process works.

When we make high-impact changes to Bevy, we need to communicate them to users (and potential users!).

> For the most part, this is done via our famously in-depth release notes.

Reviewer: It'd be nice to link to these famous release notes somewhere to show an example of what they *actually* look like. Maybe just a link to <https://bevyengine.org/news/>, or to some specific release as an example?

(Side note: If we supported filtering news posts on the Bevy website, we could do e.g. <https://bevyengine.org/news?topics=release> to directly link to a view of all release notes)

> For the most part, this is done via our famously in-depth release notes.

Author: ```suggestion
For the most part, this is done via our [famously in-depth release notes](https://bevyengine.org/news/).
```

To avoid a crunch at the end of the cycle as we *write* all of these,
Bevy asks authors (and reviewers) to write draft release notes as part of the pull requests to add high-impact features.

> ## What to put in your draft release notes

Reviewer 2: Same notes apply to this. Consider starting more explicit imperative instructions about what files to create or modify before preceding on to discussion of content.

A `template.md` file is provided in `bevyengine/bevy/working-release-notes`: copy-paste that to get started!

Release notes are intended to capture the essence of:

- what has been changed or added?
- why is this a big deal for users?
- how can they use it?

Draft release notes *do not need to be polished*: it's okay if you're not a native English speaker or aren't a wordsmith.
Editing is easy: as someone with the expertise needed to implement an awesome feature we want a rough expert's perspective that we can shape into something that reads smoothly and has a consistent voice.

> Images and videos are lovely: shiny screen shots of rendering features, diagrams, performance metrics and cool examples are all a great choice.

Reviewer: Okay final nits :P
```suggestion
Images and videos are lovely: shiny screenshots of rendering features, diagrams, performance metrics, and cool examples are all a great choice.
```

However, **do not put multimedia content in this folder**.

> We want to avoid bloating the git repo for `bevyengine/bevy`, which can cause problems for contributors (and Github).

Reviewer 3: ```suggestion
We want to avoid bloating the git repo for `bevyengine/bevy`, which can cause problems for contributors (and GitHub).
```
Nit: _GitHub_ not _Github_

Instead, drop them in your PR description and we'll collect them as we're finalizing the release notes.

## Where to put your release notes

Each major Bevy version (e.g. 0.12, or 2.0) will get its own set of release notes.
The draft release notes are organized into a folder of the same name inside of `bevyengine/bevy/working-release-notes`.

> When we release our first release candidate for a cycle, these notes are moved from `bevyengine/bevy`, and into `bevyengine/bevy-website`,

Reviewer 3: ```suggestion
When we publish our first release candidate for a cycle, these notes are moved from `bevyengine/bevy`, and into `bevyengine/bevy-website`,
```
